### DISCUSSION TITLE
MODBUS configuration for Systemair VR 400 DCV/DE

### DISCUSSION THREAD

**Discussion by simplexsolution-de** [arendst/Tasmota#16241] [Q&A]
Hi,

has anybody already implemented a modbus interface for a Systemair VR 400 DCV?

I do not get it work. (formerly I had this interface wired to RS485-USB and fhem and it
worked) But I now would like to get it connected directly to iobroker.

Can anybody help to to configure and show output data or how to test it on the console?

What I did so far is following:

1. Set the GPIO1 to TCP Tx
2. Set the GPIO3 to TCP Rx
3. Set the script to following testing set with some example datas, but I am not sure if this is correct:
>D
>B
=>sensor53 r
>M 1
+1,3,m,0,9600,MODBUS,1,10,49,65,66,67,68,81
#

I would like to get it worked with iobroker, there it already exists the setup.

The documentation of the Systemair says as followed:

**2 Transmission modes**
Modbus RTU.
Supported function codes:
1: Read Coils
2: Read Discrete Input
3: Read Holding Register
4: Read Input Register
5: Write Single Coil
6: Write Single Register
15: Write Multiple Coils
16: Write Multiple Registers

**3 Physical layer**
Two wire RS-485.
The supported communication parameters are:
9600 Bd or 19200 Bd.
No parity, even parity or odd parity.

**4 Address**
Slave address 1 to 247.
The following exception codes are generated by the slave:
1. ILLEGAL FUNCTION: when using any of the functions code not listed above.
2. ILLEGAL DATA ADDRESS: when addressing any register, coil or digital input higher than the highest
address or lower than the lowest address.
3. ILLEGAL DATA VALUE: when the format of the Modbus message is faulty (e.g. wrong message length).
7 Modbus variable list
7.1 Registers for fan control
Name Register
address
R or R/W NVM Access
(Reg./Coil)
Description/remarks
REG_FAN_SPEED_LEVEL 101 R/W Y1 Reg 0: Fans off
1: Fans on low speed
2: Fans on normal speed
3: Fans on high speed
4: Auto mode

**7 Modbus variable list (examples)**
the fully documentation can be found here:

https://shop.systemair.com/upload/assets/MODBUS_FOR_RESIDENTIAL_D24810_USER_MANUAL_EN__A007_.PDF?03eb52f3

REG_FAN_SF_FLOW_LOW 102 R/W Y1 Reg Supply air fan speed for low speed
REG_FAN_EF_FLOW_LOW 103 R/W Y1 Reg Extract air fan speed for low speed
REG_FAN_SF_FLOW_NOM 104 R/W Y1 Reg Supply air fan speed for nominal speed
REG_FAN_EF_FLOW_NOM 105 R/W Y1 Reg Extract air fan speed for nominal speed
REG_FAN_SF_FLOW_HIGH 106 R/W Y1 Reg Supply air fan speed for high speed
REG_FAN_EF_FLOW_HIGH 107 R/W Y1 Reg Extract air fan speed for high speed
REG_FAN_FLOW_UNITS 108 R/W Y Reg 0: l/s1: m3/h
REG_FAN_SF_PWM 109 R Reg 0..100: 0-10V
REG_FAN_EF_PWM 110 R Reg 0..100: 0-10V
REG_FAN_SF_RPM 111 R Reg Rotations per minute
REG_FAN_EF_RPM 112 R Reg Rotations per minute

**Comment by jimslim2** [comment]
Do you get any steps further regarding this configuration? I am also dealing with configuration problems, with no luck so far.

**Comment by benoitdepaire** [comment]
Same here but for the Systemair VR 700. If you find a way to get it working, please share. I'll do the same

**Comment by hJcRox** [comment]
Hi
I am trying to get modbus working for a SystemAir VSR300. 
For your information I found this modbus spec https://shop.systemair.com/upload/assets/SAVE_MODBUS_VARIABLE_LIST_20210301_REV36.PDF

But it is not working for me. As I have the unit connected with wifi via a SAVE connect unit with a built in web server, I am now using that one instead. It has a kind of REST API that can be used, with register ids that is approx. as in the linked specification.
